### PR TITLE
[flutter_tools] convert devtools URL to a better format

### PR DESCRIPTION
**Pull request by jonahwilliams**
Prevents URI escaping from being run over the address in the query parameter.

https://github.com/flutter/devtools/issues/3158

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart [hunk 1]
```diff
@@ -288,3 +288,21 @@ class NoOpDevtoolsHandler implements ResidentDevtoolsHandler {
...
+/// Convert a [URI] with query parameters into a display format instead
+/// of the default URI encoding.
+@visibleForTesting
+String urlToDisplayString(Uri uri) {
+  final StringBuffer base = StringBuffer(uri.replace(
+    queryParameters: <String, String>{},
+  ).toString());
+  String fencepost = '';
+  for (final MapEntry<String, String> entry in uri.queryParameters.entries) {
+    base.write(fencepost);
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/resident_devtools_handler.dart:300]: Interesting; I haven't seen strings joined this way before. You could achieve the same effect with `params.keys.map((key) => '$key=${params[key]}').join('&')` ?

**Review by devoncarew — APPROVED** [review]
lgtm (I assume you verified that the unencoded url works - devtools can load and connect to that url)?

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:300]: Yeah that seems much simpler 😅

**Comment by jonahwilliams** [comment]
Yeah launching through windows terminal or copy paste works fine, and so does the URL. Are there other workflows that I should verify first?

**Comment by devoncarew** [comment]
> Yeah launching through windows terminal or copy paste works fine, and so does the URL. Are there other workflows that I should verify first?

No, both of those sound good.

**Pull request merged**